include/psa/crypto_values.h

```diff
@@ -535,6 +582,25 @@
 #define PSA_DH_GROUP_FFDHE6144          ((psa_dh_group_t) 0x0103)
 #define PSA_DH_GROUP_FFDHE8192          ((psa_dh_group_t) 0x0104)
 
+/** Minimum value for a vendor-defined Diffie Hellman group identifier
+ *
+ * The range for vendor-defined group identifiers is a subset of the IANA
+ * registry private use range, `0x01fc` - `0x01ff`.
```

Comment:
Unfortunately, this is a ridiculously small range whereas an implementation may well wish to define many custom groups, since many groups have been common in the wild and specific systems or protocols may define their own groups.

The TLS registry puts EC and FF groups in the same 16-bit range, but we don't have to. I propose to use some values in the IANA EC range for vendor FF groups. We can in particular use the [GREASE](https://datatracker.ietf.org/doc/draft-ietf-tls-grease/) ranges, which TLS uses for the purpose of _not_ having a corresponding cryptographic algorithm.

Reply:
From a quick glance, the proposed GREASE values that can be used in this way are:

```
0x0A0A, 0x1A1A, 0x2A2A, 0x3A3A, 0x4A4A, 0x5A5A, 0x6A6A, 0x7A7A,
0x8A8A, 0x9A9A, 0xAAAA, 0xBABA, 0xCACA, 0xDADA, 0xEAEA, 0xFAFA
```

This is still draft and subject to change. This might not affect this API element, but should modify the documentation of `psa_dh_group_t`.

1. Is the suggestion to directly list the values, as being permitted for vendor-defined additional DH groups, or indirectly by referencing the specification draft?
2. Do we want to slice up the GREASE identifiers, and reserve a set for PSA Crypto future use?
3. Is the [arbitrary] 50/50 split of the TLS support group private use range appropriate (for this and ECC)?

Reply:
My proposal was to define the GREASE values (stated by value, with the GREASE spec given as an informative rationale) as reserved for vendors in `psa_dh_group_t`.

Though given that there are only 16 of them, that's not much. We can punt PSA crypto future use until later, but I do think we need more vendor FFDH values. I guess the only solution (short of moving to a bigger type) is to take values that are already assigned to elliptic curves and declare them as vendor-specific FFDH values.

Reply:
I've added a commit the extends the recommendations for vendor defined DH groups.